frontend/src/pages/hardwareProfiles/manage/validationUtils.ts

```diff
@@ -0,0 +1,37 @@
+import { z } from 'zod';
+import { IdentifierResourceType, TolerationEffect, TolerationOperator } from '~/types';
+
+const identifierSchema = z.object({
+  displayName: z.string(),
+  identifier: z.string(),
+  resourceType: z.nativeEnum(IdentifierResourceType).optional(),
+  defaultCount: z.union([z.string(), z.number()]),
+  minCount: z.union([z.string(), z.number()]),
+  maxCount: z.union([z.string(), z.number()]).optional(),
+});
+
+const tolerationSchema = z.object({
+  key: z.string(),
+  operator: z.nativeEnum(TolerationOperator).optional(),
+  value: z.string().optional(),
+  effect: z.nativeEnum(TolerationEffect).optional(),
+  tolerationSeconds: z.number().optional(),
+});
+
+export const nodeSelectorSchema = z.record(z.string().min(1), z.string().min(1));
+
+export const manageHardwareProfileValidationSchema = z.object({
+  displayName: z.string().trim().min(1, 'Display name is required'),
+  enabled: z.boolean(),
+  identifiers: z.array(identifierSchema),
```

Comment:
We need to validate that there are at least a memory and cpu resource

Reply:
we have some of these validation utils already used in `frontend/src/pages/hardwareProfiles/utils.ts`

We should merge those validation rules into your schema. Then update the helpers in that util file to work off of your schema. Also just do general cleanup around there so it all feels unified.

Reply:
The current form works without CPU and Memory identifier, are you sure we want to add validation for having atleast CPU and meemory resource?